src/MSBuild/TerminalLogger/NodesFrame.cs

```diff
@@ -0,0 +1,135 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+using System.Text;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Logging.TerminalLogger;
+
+/// <summary>
+/// Capture states on nodes to be rendered on display.
+/// </summary>
+internal sealed class NodesFrame
+{
+    private const int MaxColumn = 120;
```

Comment:
Would love feedback on this const. 120 feels reasonably wide but I regularly have wider terminals. However it's enough to not overlap stuff ~ever in my usage that I've noticed, and still "within sight".

Reply:
what does this impact?

The majority of respondents in our survey had terminals in the 600-800x200-300 range, because they were used inside of another application.

Reply:
It is the "maximum width" of the "right justification". So "right justify unless that's more than 120 chars wide, then just put it there".

Better gif:

![right-justified-max-width](https://github.com/dotnet/msbuild/assets/3347530/6cc2ff1d-2976-46c0-88f1-33a518f313da)

Reply:
Gotcha - that seems very reasonable.

Reply:
nit: could we discover this from the terminal itself via control codes?

Reply:
Do we want to discover it from the terminal, or have a cap? Having done a 4k fullscreen terminal window with full right justification . . . I think we want a cap. I'm not tied to 120 chars as the cap though.

Reply:
Definitely something we can revisit based on feedback 👍